src/dawn/SIR/SIR.h

```diff
@@ -114,6 +115,31 @@ struct Interval {
            LowerOffset == other.LowerOffset && UpperOffset == other.UpperOffset;
   }
   bool operator!=(const Interval& other) const { return !(*this == other); }
+
+  std::pair<std::string, bool> comparison(const Interval& rhs) const {
+    std::string output;
+    if(LowerLevel != rhs.LowerLevel) {
+      output += dawn::format("[Inverval missmatch] Expected LowerLevel %i, received %i", LowerLevel,
```

Comment:
I would be consistent in the formating. So far you employed the strategy 
```
[type of error] description
  expected = ...
  recieved = ...
```

Meaning I would expect an error message like
```
[Interval mismatch] LowerLevels don't match
  left  = X
  right = Y
```